src/actions/AuthActions.js

```diff
@@ -25,38 +25,42 @@ const attemptFacebookLogin = async (dispatch) => {
 
     if (type === 'success') {
         let {data: {email, name, picture: {data: {url}}}} = await axios.get(`https://graph.facebook.com/me?fields=id,name,email,picture&access_token=${token}`);
-        // we now have access to email, name, and their profile picture
-
-        await AsyncStorage.setItem(tokens.USER_TOKEN, token);
 
         dispatch({
             type: FACEBOOK_LOGIN_SUCCESS,
-            payload: token
-        })
+            payload: {token, appReady: false}
+        });
+
+        await AsyncStorage.setItem(tokens.USER_TOKEN, token);
+
+        navigate(navKeys.SEARCH);
     } else {
         dispatch({
             type: FACEBOOK_LOGIN_FAILURE
-        })
+        });
+
+        navigate(navKeys.LOGIN);
```

Comment:
If FB login is successful, send them to the `SearchScreen`. If not, send them back to the `LoginScreen`. (Keeping in mind that at this point they are now sitting on the `AuthScreen`.